batch/sc-batch-fargate.yaml

```diff
@@ -0,0 +1,198 @@
+    Type: String
+    Default: false
+    AllowedValues:
+      - true
+      - false
+  Schedule:
+    Description: >
+      Schedule to execute the docker, can be a rate or a cron schedule. Format at
+      https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/ScheduledEvents.html
+    Type: String
+    Default: cron(0 09 ? * MON *)  # Run every Monday at 9am
+    ConstraintDescription: "Use schedule format: https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/ScheduledEvents.html"
+  SynapseAuthToken:
+    Type: String
+    NoEcho: true
+    Default: ""
+    Description: >
+      (Optional) The Synapse personal access token (PAT).
+      https://docs.synapse.org/rest/POST/personalAccessToken.html
+Resources:
+  ComputeEnvironment:
+    Type: AWS::Batch::ComputeEnvironment
+    Metadata:
+      cfn-lint:
+        config:
+          ignore_checks:
+            - E3008
+    Properties:
+      Type: MANAGED
+      State: ENABLED
+      ServiceRole: !Ref ServiceRole
+      ComputeResources:
+        Type: FARGATE
+        MaxvCpus: 1
+        Subnets:
+          - !ImportValue us-east-1-scriptrunnervpc-PrivateSubnet
+          - !ImportValue us-east-1-scriptrunnervpc-PrivateSubnet1
+          - !ImportValue us-east-1-scriptrunnervpc-PrivateSubnet2
+        SecurityGroupIds:
+          - !Ref SecurityGroup
+  ServiceRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: 'Allow'
+            Principal:
+              Service:
+                - 'batch.amazonaws.com'
+            Action:
+              - 'sts:AssumeRole'
+      ManagedPolicyArns:
+        - 'arn:aws:iam::aws:policy/service-role/AWSBatchServiceRole'
+  JobQueue:
+    Type: AWS::Batch::JobQueue
+    Properties:
+      ComputeEnvironmentOrder:
+        - ComputeEnvironment: !Ref ComputeEnvironment
+          Order: 1
+      Priority: 1
+      State: ENABLED
+  SchedulingPolicy:
+    Type: AWS::Batch::SchedulingPolicy
+    Metadata:
+      cfn-lint:
+        config:
+          ignore_checks:
+            - E3001
+    Properties:
+      FairsharePolicy:
+        ShareDecaySeconds: 5
+        ComputeReservation: 2
+  SecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      VpcId: !ImportValue us-east-1-scriptrunnervpc-VPCId
+      GroupDescription: !Sub 'SC for ${AWS::StackName}}'
+  JobDefinition:
+    Type: AWS::Batch::JobDefinition
+    Properties:
+      Type: container
+      JobDefinitionName: { Ref: "AWS::StackName" }
+      PropagateTags: True
+      PlatformCapabilities:
+        - FARGATE
+      Timeout:
+        AttemptDurationSeconds: 600
+      RetryStrategy:
+        Attempts: 1
+      ContainerProperties:
+        Command: !Ref Command
+        Image: !Ref Image
+        Environment:
+          - Name: SYNAPSE_AUTH_TOKEN
+            Value: !Ref SynapseAuthToken
```

Comment:
After talking with @zaro0508 , we think that this is the only input method for running a batch job through Service Catalog, particularly if the job requires the user to input a *scoped* access token... By taking input it was to be a stack parameter (though it could be stored somewhere else after). Other products can take secrets input after they are created, for example. If this is the only method, then we also need to restrict access to any non-admin roles to read stack parameters.

Reply:
@ahayden Did you and @zaro0508 discuss the option of having the user put the secret into SSM (secure) parameter store, e.g. using the AWS CLI?  (The batch job would then retrieve the secret from there at run time.) I'm curious to know why you feel that option is infeasible.